src/main/java/org/opensrp/service/PlanService.java

```diff
@@ -0,0 +1,73 @@
+package org.opensrp.service;
+
+/**
+ * Created by Vincent Karuri on 06/05/2019
+ */
+
+import org.apache.commons.lang.StringUtils;
+import org.opensrp.domain.PlanDefinition;
+import org.opensrp.repository.PlanRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class PlanService {
+
+    private PlanRepository planRepository;
+
+    @Autowired
+    public void setPlanRepository(PlanRepository planRepository) {
+        this.planRepository = planRepository;
+    }
+
+    public PlanRepository getPlanRepository() {
+        return planRepository;
+    }
+
+    public List<PlanDefinition> getAllPlans() {
+        return getPlanRepository().getAll();
+    }
+
+    public void addOrUpdatePlan(PlanDefinition plan) {
+        if (StringUtils.isBlank(plan.getIdentifier())) {
+            throw new IllegalArgumentException("Identifier not specified");
+        }
+        plan.setServerVersion(System.currentTimeMillis());
+        if (getPlanRepository().get(plan.getIdentifier()) != null) {
```

Comment:
Change this to query for the plan using the identifier, it should be able to add a plan that has an identifier that does not exist.

The flow should be query the identifier(PK) if its found update otherwise insert